festim/subdomain/interface.py

```diff
@@ -0,0 +1,87 @@
+import festim as F
+import dolfinx
+from dolfinx.cpp.fem import compute_integration_domains
+import numpy as np
+
+
+class Interface:
+    id: int
+    subdomains: tuple[F.VolumeSubdomain, F.VolumeSubdomain]
+    parent_mesh: dolfinx.mesh.Mesh
+    mt: dolfinx.mesh.MeshTags
+    restriction: list[str, str] = ("+", "-")
+    padded: bool
+
+    def __init__(
+        self,
+        id: int,
+        subdomains: list[F.VolumeSubdomain],
+        penalty_term: float = 10.0,
+    ):
+        """Class representing an interface between two subdomains.
+
+        Args:
+            id (int): the tag of the interface subdomain in the parent meshtags
+            subdomains (list[F.VolumeSubdomain]): the subdomains sharing this interface
+            penalty_term (float, optional): Penalty term in the Nietsche DG formulation.
+                Needs to be "sufficiently large". Defaults to 10.0.
+        """
+        self.id = id
+        self.subdomains = tuple(subdomains)
+        self.penalty_term = penalty_term
+
+    def pad_parent_maps(self):
+        """Workaround to make sparsity-pattern work without skips"""
+
+        integration_data = compute_integration_domains(
+            dolfinx.fem.IntegralType.interior_facet,
+            self.parent_mesh.topology._cpp_object,
+            self.mt.find(self.id),
+            self.mt.dim,
+        ).reshape(-1, 4)
+        for i in range(2):
+            # We pad the parent to submesh map to make sure that sparsity pattern is correct
+            mapped_cell_0 = self.subdomains[i].parent_to_submesh[integration_data[:, 0]]
+            mapped_cell_1 = self.subdomains[i].parent_to_submesh[integration_data[:, 2]]
+            max_cells = np.maximum(mapped_cell_0, mapped_cell_1)
+            self.subdomains[i].parent_to_submesh[integration_data[:, 0]] = max_cells
+            self.subdomains[i].parent_to_submesh[integration_data[:, 2]] = max_cells
+            self.subdomains[i].padded = True
+
+    def compute_mapped_interior_facet_data(self, mesh):
+        """
+        Compute integration data for interface integrals.
+        We define the first domain on an interface as the "+" restriction,
+        meaning that we must sort all integration entities in this order
```

Comment:
At the moment these are still quite a black box, could maybe provide some links to documentation and or add some more details to the doc strings